packages/relay-compiler/bin/RelayCompilerBin.js

```diff
@@ -132,14 +161,34 @@ Ensure that one such file exists in ${srcDir} or its parents.
   const useWatchman = options.watchman && (await WatchmanClient.isAvailable());
 
   const schema = getSchema(schemaPath);
+
+  const languagePlugin = getLanguagePlugin(options);
+
+  const extensions = options.extensions || languagePlugin.inputExtensions;
+
+  const sourceModuleParser = RelaySourceModuleParser(
+    languagePlugin.findGraphQLTags,
+  );
+
+  const artifactDirectory = options.artifactDirectory
+    ? // $FlowFixMe artifactDirectory can’t be null/undefined at this point
+      path.resolve(process.cwd(), options.artifactDirectory)
+    : null;
+
+  const isGeneratedDirectory = artifactDirectory || '__generated__';
```

Comment:
nit: maybe this would be better named as `generatedDirectoryName`?

Reply:
👍